### ISSUE TITLE
use strict/warnings does not pass. [rt.cpan.org #90340]

### ISSUE THREAD

**Issue by toddr** [Perl/perl5#17495]
Migrated from [rt.cpan.org#90340](https://rt.cpan.org/Ticket/Display.html?id=90340) (status was 'new')

Requestors:
* Michael.J.Mestnik@usps.gov

From michael.j.mestnik@usps.gov on 2013-11-13 20:01:24
:
```
Variable spelling error line 759.
Lots of scoping issues(my used instead of our, ?local used at all?).
Typecast issues, * used instead of @ in my.

$DIAGNOSTIC  <---  This can't work the way you intend(can it?), A variable defined with a value that is cleared when that first copy goes out of scope?  Try using sub a{return do {$Tie::File::DIAGNOSTIC=1;... }continue{$Tie::File::DIAGNOSTIC=0}}; "elsewhere"; die if($Tie::File::DIAGNOSTIC);

Others?  Try testing with strict/warning in each package.

Cheers.



```

**Comment by haarg** [comment]
Doesn't seem like a bug, the module just isn't written in a "modern" style.

**Comment by toddr** [comment]
True but its a module p5p maintains. It seems like something that should have a minimum set of standards

**Comment by jkeenan** [comment]
Original poster:  What `$VERSION` of Tie::File were you using when you originally filed this ticket in 2013?  (I suspect 0.99 but we could use confirmation to address your complaints.)

Later contributors:  If we don't hear from the OP within, say, 30 days, it will be difficult to take action on this ticket as we won't have enough details to proceed.  If that is the case, then we should close this ticket and re-open a new one with a more specific subject like _"use strict and warnings throughout."_

Thank you very much.
Jim Keenan

**Comment by toddr** [comment]
```
$>perl -cw -Mstrict lib/Tie/File.pm 
Variable "@OFF" is not imported at lib/Tie/File.pm line 872.
Variable "@OFF" is not imported at lib/Tie/File.pm line 876.
Variable "@OFF" is not imported at lib/Tie/File.pm line 885.
Variable "@OFF" is not imported at lib/Tie/File.pm line 900.
Variable "@OFF" is not imported at lib/Tie/File.pm line 904.
Variable "@H" is not imported at lib/Tie/File.pm line 1231.
Variable "@H" is not imported at lib/Tie/File.pm line 1234.
Variable "@H" is not imported at lib/Tie/File.pm line 1235.
Variable "@H" is not imported at lib/Tie/File.pm line 1237.
Variable "@H" is not imported at lib/Tie/File.pm line 1238.
Variable "@H" is not imported at lib/Tie/File.pm line 1239.
Variable "@H" is not imported at lib/Tie/File.pm line 1239.
Variable "@H" is not imported at lib/Tie/File.pm line 1243.
Variable "@H" is not imported at lib/Tie/File.pm line 1248.
Global symbol "$DIAGNOSTIC" requires explicit package name (did you forget to declare "my $DIAGNOSTIC"?) at lib/Tie/File.pm line 199.
Global symbol "%newkey" requires explicit package name (did you forget to declare "my %newkey"?) at lib/Tie/File.pm line 764.
Global symbol "@OFF" requires explicit package name (did you forget to declare "my @OFF"?) at lib/Tie/File.pm line 872.
Global symbol "@OFF" requires explicit package name (did you forget to declare "my @OFF"?) at lib/Tie/File.pm line 876.
Global symbol "@OFF" requires explicit package name (did you forget to declare "my @OFF"?) at lib/Tie/File.pm line 885.
Global symbol "@OFF" requires explicit package name (did you forget to declare "my @OFF"?) at lib/Tie/File.pm line 900.
Global symbol "@OFF" requires explicit package name (did you forget to declare "my @OFF"?) at lib/Tie/File.pm line 904.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1231.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1234.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1235.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1237.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1238.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1239.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1239.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1243.
Global symbol "@H" requires explicit package name (did you forget to declare "my @H"?) at lib/Tie/File.pm line 1248.
Global symbol "$DIAGNOSTIC" requires explicit package name (did you forget to declare "my $DIAGNOSTIC"?) at lib/Tie/File.pm line 1301.
BEGIN not safe after errors--compilation aborted at lib/Tie/File.pm line 1460.
```

**Comment by toddr** [comment]
This should be addressed by PR #17511
